lib/validate.js

```
@@ -392,6 +438,16 @@ function validateCreateContainer(req, res, next) {
             'HostConfig.PortBindings');
         assertOptionalObject(container.HostConfig.LogConfig,
             'HostConfig.LogConfig');
+        assertOptionalString(container.HostConfig.NetworkMode,
+            'HostConfig.NetworkMode');
+
+        if (container.HostConfig.NetworkMode) {
+            assertNetworkMode(container.HostConfig.NetworkMode,
+                'HostConfig.NetworkMode');
+        }
+
+        // ensures mutually-compatible networking params.
+        assertNetworkingParams(container.HostConfig);
```

Ah, so `assertNetworkParams` should throw instead of return.